### ISSUE TITLE
Support for global introspection ( @import )

### ISSUE THREAD

**Issue by mehcode** [stylelint/stylelint#1199]
How possible (and desirable) is it to get stylelint to understand and follow `@import`. I guess that's more of a scss/less feature than straight css but it sounds like something that'd have to be at least exposed here (and then used in `stylelint-scss`).

I'd like to be able to have lints that are project-wide. For instance, `no-duplicate-selectors`, is a great example of something that is useful but would be infinitely more so if it could do project-wide inspection.

If this is something that we want I can take a crack at implementing.

**Comment by davidtheclark** [comment]
It's a good idea. All depends on how it can be pulled off.

I think we definitely don't want stylelint itself to follow `@import`s. That's a tricky task that should be handled by an outside module. `postcss-import` should work, right? So could we possibly add an option that would make `postcss-import` run on the code before stylelint? Is that something like what you were thinking?

The way that people would invoke stylelint when using this feature would have to be different: you'd want to only lint the main file that imports all the rest, not tell stylelint to check every file.

**Comment by mehcode** [comment]
@davidtheclark Actually, just adding `postcss-import` before stylelint in the postcss invocation works great. There is only 1 problem that I've found (blank lines get removed): https://github.com/postcss/postcss-import/issues/206

Perhaps we should document this usage as there are a good number of rules that work better this way (in my opinion).

**Comment by davidtheclark** [comment]
That works if you use stylelint within a PostCSS pipeline. But I was thinking an option might help CLI and Node API users.

**Comment by mehcode** [comment]
That might be nice. It might also be nice to figure out a clean way to make it work for editor plugins (sublime, atom, etc.), if possible.

**Comment by MoOx** [comment]
Adding stylelint before postcss-import will not lint `@import`ed files. That's why postcss-import have a plugin option https://github.com/postcss/postcss-import#plugins
Eg: 

```js
require("stylelint")(),
require("postcss-import")({ plugins: [ require("stylelint")() ] }),
```

**Comment by davidtheclark** [comment]
@MoOx But using postcss-import *before* stylelint *will* lint `@import`ed files, right?

**Comment by MoOx** [comment]
Yeah but maybe not in the right context (filename or something?). Eg: will bem-linter work this way?

**Comment by jeddy3** [comment]
I think this has implications for editor plugins i.e. are they expected to follow `@import` too? If so, won't that significantly slow down their response time?

If this feature isn't applicable to editor plugins then perhaps there another way of achieving the same goal i.e. a separate linting task, with a specialised config, that only checks the *transformed CSS*. For example, not only could you check for duplicate selectors across the whole code base, but you can also check for things specific to the transformed CSS e.g. making sure only data URIs are used. There is no need for whitespace rules (unless your checking the minifier transform has worked). I think this works around the "missing empty lines" and (possible) bem-linter limitations.

This seems like the best balance between having a safety net and making work for ourselves.

**Comment by davidtheclark** [comment]
Creating a post-complication linting task seems like the right solution to me. It's already possible, is not a huge amount of added work, and does not add complexity to stylelint.

**Comment by alexreg** [comment]
@davidtheclark What do you mean "post-complication"? What's the best way to achieve this currently?
